src/hotel_reservations/utils.py

```diff
@@ -0,0 +1,124 @@
+import json
+import logging
+import os
+from typing import Any, Optional
+
+import requests
+import yaml
+from databricks.sdk import WorkspaceClient
+from pyspark.ml.evaluation import RegressionEvaluator
+from pyspark.sql import DataFrame
+
+from hotel_reservations.types.project_config_types import ProjectConfig
+
+
+def get_api_token_from_secrets(scope: str, key: str) -> str:
+    """Retrieves Databricks secrets with dbutils, this is used to retrieve the Databricks volume path and the Databricks user directory path. When no profile is assigned to the WorkspaceClient, the DEFAULT profile is used.
+
+    Args:
+        scope (str): Name of the Databricks secret scope
+        key (str): Name of the secret
+
+    Returns:
+        str: The value of the secret
+    """
+    w = WorkspaceClient(profile=os.environ.get("DATABRICKS_PROFILE", None))
+    secret = w.dbutils.secrets.get(scope=scope, key=key)
+    return secret
+
+
+def open_config(path: str, scope: str) -> ProjectConfig:
+    """Opens the project config file based on the path given, the user_dir_path and volume_whl_path are retrieved from Databricks dbutils secrets. Make sure you've read the documentation and followed the steps in the create_databricks_secrets notebook before calling this function.
+
+    Args:
+        path (str): Path to the project config yaml file
+        scope (str): Name of the Databricks secret scope
+
+    Raises:
+        FileNotFoundError: When the given path is not found
+        ValueError: When the file is not valid YAML
+
+    Returns:
+        ProjectConfig: Project configuration file containing the catalog and schema where the data resides. Moreover, it contains the model parameters, numerical features, categorical features and the target variables.
+    """
+    try:
+        with open(path, "r") as file:
+            config = yaml.safe_load(file)
+        logging.info("Configuration file loaded successfully from project_config.yaml")
+    except FileNotFoundError as e:
+        msg = f"Configuration file not found at '{path}'. Ensure it exists: {e}"
+        logging.error(msg)
+        raise FileNotFoundError(msg) from e
+    except yaml.YAMLError as e:
+        msg = f"Failed to parse configuration file at '{path}'. Ensure it's valid YAML: {e}"
+        logging.error(msg)
+        raise ValueError(msg) from e
+
+    config["user_dir_path"] = get_api_token_from_secrets(scope, "user_dir_path")
```

Comment:
see comment above, confusing name, also feels like that function is not needed at all. makes your code confusing and harder to understand (I have to lookup and understand what your function does while it does not do anything special what requires a function)